HW04.Rmd

```diff
@@ -29,32 +29,101 @@ Knitr is part of the knitr package. Allows you to change column names and add ca
+    kable(format = "markdown")
 ```
 
 
 ### Generating a bar chart with human-readable labels on the x-axis. That is, each month should be labeled "Jan", "Feb", "Mar" (full or abbreviated month names are fine), not `1`, `2`, `3`.
 
 ```{r month_chart}
-
+#Makes abbreviated months leveled factors for proper display via ggplot
+gun_deaths_by_month$abMonth <- factor(gun_deaths_by_month$abMonth, 
+                                      levels = c(month.abb[12:1]))
+#Plots data of deaths by month
+set.seed(3)
+gun_deaths_by_month  %>% 
+    ggplot(mapping = aes(x = abMonth, y = deaths_month)) +
+    geom_bar(stat = "summary", fun = "mean", alpha = 0.7, width = 0.4) +
+    geom_errorbar(stat = "summary", 
+                  mapping = aes(ymin = deaths_month - sd, 
+                                ymax = deaths_month + sd, width = 0.2)) +
+    scale_y_continuous() + coord_flip() +
+    xlab("Month") + ylab("American Gun Deaths\n(average, standard deviation)") +
+    ggtitle("American Gun Deaths by Month (2012-2014)")
 ```
 
 ## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
 
 ```{r intent}
-
+#Selects and enumerates death by intent, trimmed unavailable datum
+gun_deaths_by_intent <- gun_deaths %>%
+    select(intent) %>%
+    count(intent, name = "deaths_intent") %>%
+    filter(intent != "Not Available") %>%
+    arrange(desc(deaths_intent))
+
+#Plots data of deaths by determined intent
+gun_deaths_by_intent %>%
+ggplot(mapping = aes(x = reorder(intent, -deaths_intent), y = deaths_intent)) +
+    geom_bar(stat = "identity", alpha = 0.7, width = 0.7) +
+    xlab("Determined Intent of Gun Death") + ylab("Number of Gun Deaths") +
+    ggtitle("American Gun Deaths by Intent (2012-2014)")
 ```
 
 ## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
 
 ```{r age_&_sex}
-
+#Selects age and sex variables from gun_deaths
+gun_deaths_by_age_sex <- gun_deaths %>%
+    select(age, sex) %>%
+    filter(age != "Not Available") %>% 
+    transform(age = as.numeric(age)) %>% 
+    group_by(sex) %>%
+    mutate(mean_age = mean(as.numeric(age))) %>%
+    add_count(sex, name = "deaths_sex") %>%
+    ungroup()
+
+#Boxplot of age vs sex
+gun_deaths_by_age_sex %>% 
+    ggplot(mapping = aes(x = sex, y = age, color = sex)) +
+    geom_boxplot(fill = NA, outlier.shape = NA, width = 0.595) + 
+    geom_point(size = 0, position = position_jitter(width = 0.3), alpha = 0.1) +
+    scale_color_brewer(palette = "Set2") +
+    xlab("Sex of Victims") + ylab("Age of Victims") +
+    ggtitle("Sex and Age of Gun Death Victims in America (2012-2014)") +
+    scale_x_discrete(labels=c("F" = "Female", "M" = "Male")) +
+    theme(legend.position = "none", axis.ticks.x = element_blank(),
+          plot.title = element_text(hjust = 0.5))

```

Comment:
When I graph this, I see these bands of empty space where I feel like data should be. When I look closely, I'm pretty sure I see a few data points in those gaps, which is confusing to me. I don't know if this is a problem with the use of position_jitter with this graph, or something else.